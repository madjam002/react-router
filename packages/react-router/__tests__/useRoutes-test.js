import * as React from 'react';
import { create as createTestRenderer } from 'react-test-renderer';
import { MemoryRouter as Router, useRoutes } from 'react-router';

describe('useRoutes', () => {
  function Home() {
    return <h1>Home</h1>;
  }

  function About() {
    return <h1>About</h1>;
  }

  it('returns the matching element from a route config', () => {
    function RoutesRenderer({ routes }) {
      return useRoutes(routes);
    }

    let routes = [
      { path: 'home', element: <Home /> },
      { path: 'about', element: <About /> }
    ];

    let renderer = createTestRenderer(
      <Router initialEntries={['/home']}>
        <RoutesRenderer routes={routes} />
      </Router>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('accepts `basename` as optional parameter', () => {
    function RoutesRenderer({ routes }) {
      return useRoutes(routes, {
        "basename": "/parent",
      });
    }

    let routes = [
      { path: 'home', element: <Home /> },
      { path: 'about', element: <About /> }
    ];

    let renderer = createTestRenderer(
      <Router initialEntries={["/parent/home"]}>
        <RoutesRenderer routes={routes} />
      </Router>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('accepts `location` as optional parameter', () => {
    let location = {
      "pathname": "/about",
    };

    function RoutesRenderer({ routes }) {
      return useRoutes(routes, {
        "location": location,
      });
    }

    let routes = [
      { path: 'home', element: <Home /> },
      { path: 'about', element: <About /> }
    ];

    let renderer = createTestRenderer(
      <Router initialEntries={["/home"]}>
        <RoutesRenderer routes={routes} />
      </Router>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
