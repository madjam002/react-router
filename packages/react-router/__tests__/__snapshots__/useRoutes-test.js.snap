// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useRoutes accepts \`basename\` as optional parameter 1`] = `
<h1>
  Home
</h1>
`;

exports[`useRoutes accepts \`location\` as optional parameter 1`] = `
<h1>
  About
</h1>
`;

exports[`useRoutes returns the matching element from a route config 1`] = `
<h1>
  Home
</h1>
`;
